apps/designer/app/canvas/shared/instance.ts

```diff
@@ -38,10 +38,10 @@ import {
 import { useMeasure } from "~/shared/dom-hooks";
 
 export const usePopulateRootInstance = (tree: Tree) => {
-  const [, setRootInstance] = useRootInstance();
-  useEffect(() => {
-    setRootInstance(tree.root);
-  }, [tree, setRootInstance]);
+  // It is only set once when the canvas is first loaded.
+  if (rootInstanceContainer.value === undefined) {
+    rootInstanceContainer.value = tree.root;
```

Comment:
yes, this is terribly wrong and I didn't realize it from ssr perspective, all of those containers and hooks from react-nano-state should be created during ssr, not reused

Reply:
this one on the other hand was actually a problem, because it would not clean up on every ssr render, I used useRef now, still not great, but I have a plan for a future migration